### PR TITLE
feat: add new CheckSecrets dripcheck

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Adds a new dripcheck called CheckSecrets that acts somewhat like a timelock contract. Slightly more useful than a traditional timelock because it can be triggered by any account. Very useful if the account is actively being drained which would require using a third-party service like flashbots to circumvent.

### CODE REVIEW

**Review by Inphi — APPROVED** [review]

**Comment by smartcontracts** [comment]
Going to wait for review from @clabby before merging

**Review by clabby — APPROVED** [review]

File: packages/contracts-bedrock/src/periphery/drippie/dripchecks/CheckSecrets.sol
Comment (maurelian) [inline packages/contracts-bedrock/src/periphery/drippie/dripchecks/CheckSecrets.sol]: would it be beneficial to index this arg, and also emit the unhashed secret here?
```suggestion
    event SecretRevealed(bytes32 indexed secretHash, bytes secret);
```

File: packages/contracts-bedrock/src/periphery/drippie/dripchecks/CheckSecrets.sol
Comment (maurelian) [inline packages/contracts-bedrock/src/periphery/drippie/dripchecks/CheckSecrets.sol:1]: throughout this file, I found it slightly confusing that the word `secret` is used to refer to both the secret `bytes` value, and its hashed `bytes32` value. 

Would it be too unwieldy to use `secretHash` to refer to the `bytes32` values?

File: packages/contracts-bedrock/src/periphery/drippie/dripchecks/CheckSecrets.sol
Comment (smartcontracts) [reply packages/contracts-bedrock/src/periphery/drippie/dripchecks/CheckSecrets.sol]: Yeah sure, why not. I'll add it.

File: packages/contracts-bedrock/src/periphery/drippie/dripchecks/CheckSecrets.sol
Comment (smartcontracts) [reply packages/contracts-bedrock/src/periphery/drippie/dripchecks/CheckSecrets.sol:1]: Yep makes sense to me.

**Pull request merged**
